### PR TITLE
OSD-19453: Drain reboot worker nodes

### PR DESCRIPTION
**Pull request by faldanarh**
### What type of PR is this?

Feature

### What this PR does / Why we need it?

This PR is necessary to add a new managed script for draining/rebooting worker nodes.

### Which Jira/Github issue(s) does this PR fix?

_Resolves #_ https://issues.redhat.com/browse/OSD-19453

### Pre-checks (if applicable)

- [X] Validated the changes in a cluster
- [X] Included documentation changes with PR

### CODE REVIEW

File: scripts/CEE/drain-reboot-nodes/script.sh [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+check_worker(){
+
+    echo "Checking if \"${WORKER}\" is a worker node..."
+    
+    if (oc get nodes -l node-role.kubernetes.io/worker=,node-role.kubernetes.io/infra!= -oname | grep "${WORKER}") &> /dev/null; then
+        echo "[OK] \"${WORKER}\" is a worker node. Proceeding with draining..."
+        echo
+    else
+        echo "[Error] \"${WORKER}\" is not a worker node. Exiting script..."
+        exit 1
```
Comment (bmeng) [inline scripts/CEE/drain-reboot-nodes/script.sh:41]: `exit 1` will call the job failed, but as it is exiting as expected, maybe we do not exit as error?

Comment (feichashao) [reply scripts/CEE/drain-reboot-nodes/script.sh:41]: I think `exit 1` makes sense here, as we want to mark the job as failed, and the user can check the logs to know the error.

File: scripts/CEE/drain-reboot-nodes/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        - "patch"
+        - "list"
+        - "watch"
+        - "create"
+        - "delete"
+      apiGroups:
+        - ''
+      resources:
+        - "pods"
+        - "pods/attach"
```
Comment (feichashao) [inline scripts/CEE/drain-reboot-nodes/metadata.yaml:30]: The debug pod in the script is running in namespace `default`, we can limit the permission to namespace `default`. 
```
oc -n default debug node/$WORKER
```

example:  https://github.com/openshift/managed-scripts/blob/651039bbcc4a7aebba47f0a165a4257f177865b0/scripts/SREP/example/metadata.yaml#L16

Comment (faldanarh) [reply scripts/CEE/drain-reboot-nodes/metadata.yaml]: I have added that as recommended.

File: scripts/CEE/drain-reboot-nodes/script.sh [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    fi
+
+}
+
+## Function which drains the node bypassing the PDB check
+drain_worker(){
+
+    echo "Draining node \"${WORKER}\"..."
+    oc adm drain "${WORKER}" --ignore-daemonsets --delete-emptydir-data --force --disable-eviction
```
Comment (feichashao) [inline scripts/CEE/drain-reboot-nodes/script.sh:50]: I overlooked this when suggesting the RBAC changes.  I suppose this command still need get/list/delete pod permission cluster-wide?  I haven't tested it though.   Could you help to test it when possible?

**Comment by faldanarh** [comment]
Job logs:

```
Job started at 2023-12-13 07:36:41
..................................

Checking if "ip-10-0-29-10.us-west-2.compute.internal" is a worker node...
[OK] "ip-10-0-29-10.us-west-2.compute.internal" is a worker node. Proceeding with draining...

Draining node "ip-10-0-29-10.us-west-2.compute.internal"...
node/ip-10-0-29-10.us-west-2.compute.internal already cordoned
Warning: ignoring DaemonSet-managed Pods: openshift-cluster-csi-drivers/aws-ebs-csi-driver-node-r2p7x, openshift-cluster-node-tuning-operator/tuned-ll6xw, openshift-dns/dns-default-9277l, openshift-dns/node-resolver-j289h, openshift-image-registry/node-ca-dsss9, openshift-ingress-canary/ingress-canary-lrnzq, openshift-machine-config-operator/machine-config-daemon-xpnvg, openshift-monitoring/node-exporter-b6fvc, openshift-multus/multus-additional-cni-plugins-7xzdf, openshift-multus/multus-v7z77, openshift-multus/network-metrics-daemon-5zkz5, openshift-network-diagnostics/network-check-target-5ghmq, openshift-ovn-kubernetes/ovnkube-node-zpfdq, openshift-security/splunkforwarder-ds-rbm6m
pod/collect-profiles-28374180-lzk6g deleted
pod/collect-profiles-28374195-dtv2t deleted
pod/configure-alertmanager-operator-74c5658b85-8k4cd deleted
pod/monitoring-plugin-8567484b97-9rl8w deleted
pod/image-registry-67cf7f5dc5-scdq8 deleted
node/ip-10-0-29-10.us-west-2.compute.internal drained
[OK] Node "ip-10-0-29-10.us-west-2.compute.internal" drained successfully.

Rebooting node "ip-10-0-29-10.us-west-2.compute.internal"...
Starting pod/ip-10-0-29-10us-west-2computeinternal-debug-xk8r5 ...
To use host binaries, run `chroot /host`
Pod IP: 10.0.29.10
If you don't see a command prompt, try pressing enter.
sh: cannot set terminal process group (112858): Inappropriate ioctl for device
sh: no job control in this shell
sh-5.1#     reboot
sh-5.1# exit

Removing debug pod ...

Checking node reboot...
Rebooting...
Rebooting...
Rebooting...
Rebooting...
Rebooting...
Rebooting...
Rebooting...
Rebooting...
Rebooting...
Rebooting...
Rebooting...
Rebooting...
Rebooting...
Rebooting...
Rebooting...
Node Rebooted!

..................................
Job finished at 2023-12-13 07:36:41

```

File: scripts/CEE/drain-reboot-nodes/script.sh (hunk 3 above)
Comment (faldanarh) [reply scripts/CEE/drain-reboot-nodes/script.sh:50]: Done.

**Pull request merged**